src/components/sales/AddEditProduct.vue

```diff
@@ -331,9 +331,13 @@ export default defineComponent({
         if (this.product.productVariants.length > 1) {
           const deleteProductVariantID = this.product.productVariants[index].id;
           // Delete Variant
-          this.deleteProductVariant(deleteProductVariantID);
-          this.updateDeleteLastVarient = '';
-          this.product.productVariants.splice(index, 1);
+          if (deleteProductVariantID) {
+            this.deleteProductVariant(deleteProductVariantID);
+            this.updateDeleteLastVarient = '';
+            this.product.productVariants.splice(index, 1);
```

Comment:
why do we have to manually update the list. Should be updated automatically as the state changes after delete and getter should have updated list.

Reply:
@ammarhaiderak Then, we have manually fetch the product after each insertion and deletion, that's why.